src/components/Card.js

```diff
@@ -1,29 +1,53 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick,
+    handleLikeClick) {
+    this.name = data.name;
+    this.link = data.link; 
+    this._id = data._id; 
+    this.isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
-  }
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
+  } 
 
+  getCardId() {
+    return this._id;
+  }
+  
   _setEventListeners() {
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
+      this.handleLikeIcon(this);
```

Comment:
Needs correction: here you should use the provided handler ` this._handleLikeClick`